lib/register_client.rb

```diff
@@ -109,6 +111,24 @@ def get_records_with_history_for_entry_type(since_entry_number, entry_type)
       records_with_history
     end
 
+    def get_register_proof(register, phase)
+      @register_proof = JSON.parse(RestClient.get("https://#{register}.#{phase}.openregister.org/proof/register/merkle:sha-256"))
+    end
+
+    def validate_register_integrity(rsf, current_entry_number)
+      rsf_lines = rsf.split("\n")
+      latest_root_hash = rsf_lines[rsf_lines.length - 1].split("\t")[1]
+      register_proof = get_register_proof(@register, @phase)
```

Comment:
@karlbaker02 This gets the current register proof of the remote register (i.e. the one at *.register.gov.uk). Should it instead be comparing to the register proof of the register held locally? Meaning that it needs to get the root hash from the data store?

Reply:
I think you're right. `DataStore` however currently has no support for getting the root hash of a register, nor do we store it. I'll add a card to address this.

Reply:
Yes I guess we can store it by telling it what it's current root hash is from the last RSF file consumed.